client/data/hosting/use-cache.ts

```diff
@@ -0,0 +1,147 @@
+import {
+	useQuery,
+	useMutation,
+	UseMutationOptions,
+	useQueryClient,
+	useIsMutating,
+} from '@tanstack/react-query';
+import { useCallback } from 'react';
+import wp from 'calypso/lib/wp';
+
+export const EDGE_CACHE_ENABLE_DISABLE_NOTICE_ID = 'edge-cache-enable-disable-notice';
+export const USE_EDGE_CACHE_QUERY_KEY = 'edge-cache-key';
+export const TOGGLE_EDGE_CACHE_MUTATION_KEY = 'set-edge-site-mutation-key';
+export const CLEAR_EDGE_CACHE_MUTATION_KEY = 'clear-edge-site-mutation-key';
+
+interface ClearEdgeCacheMutationVariables {
+	name: string;
+}
+
+interface SetEdgeCacheMutationVariables {
+	siteId: number;
+	active: boolean;
+}
+
+interface MutationResponse {
+	message: string;
+}
+
+interface MutationError {
+	code: string;
+	message: string;
+}
+
+export const getEdgeCacheStatus = async ( siteId: number ) => {
+	const response = await wp.req.get( {
```

Comment:
Why not return here?

Reply:
https://github.com/Automattic/wp-calypso/pull/85902/commits/77e7ecbc098bfc4b330efeb823574e0651cd308f